src/scheduler/context.rs

```diff
@@ -558,6 +558,11 @@ impl RContextForwardableStuff<'_> {
         self.future_events.append(&mut other.future_events);
```

Comment:
You could now make this merge function use the other to remove code duplication.